src/shard_log.c

```diff
@@ -222,8 +223,10 @@ get_new_page_for_thread(shard_log             *log,
 }
 
 int
-shard_log_write(log_handle *logh, slice key, message msg, uint64 generation)
+shard_log_write(log_handle *logh, key key, message msg, uint64 generation)
 {
+   debug_assert(key_is_user_key(key));
```

Comment:
That's a reasonable question.  However, the general reason I want to use keys everywhere is for future support for huge keys (i.e. blobs).

When blob support for huge keys lands, the flag indicating whether the key is inline or a blob will be stored in the key structure.  The slice in the key will hold either the inline key or the blob descriptor (i.e. the array of pointers to the extents and pages used by the blob).  And this function will need to know which kind of key it is dealing with.  So it will need a key, not a slice.